src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -362,23 +369,50 @@ private PaymentTransactionInfoPlugin executeInitialTransaction(final Transaction
                                                                    final Currency currency,
                                                                    final Iterable<PluginProperty> properties,
                                                                    final CallContext context) throws PaymentPluginApiException {
-        final Account account = getAccount(kbAccountId, context);
-
-        final Long transactionAmount = (amount == null ? null : amount.longValue());
-        final PaymentData paymentData = buildPaymentData(account, kbPaymentId, kbTransactionId, kbPaymentMethodId, currency, properties, context);
-        final OrderData orderData = buildOrderData(account, properties);
-        final UserData userData = buildUserData(account, properties);
-        final String termUrl = PluginProperties.findPluginPropertyValue(PROPERTY_TERM_URL, properties);
-        final SplitSettlementData splitSettlementData = null;
-        final DateTime utcNow = clock.getUTCNow();
+        killbillMetrics.markMeter(transactionType.toString());
```

Comment:
Maybe we can split the problem in two.

Could we actually do something in killbill-platform (e.g. in [ContextClassLoaderHelper](https://github.com/killbill/killbill-platform/blob/a816f514189bcbb0ced80d165deae82a7a8d7699/osgi/src/main/java/org/killbill/billing/osgi/ContextClassLoaderHelper.java)), such as instrumenting timing and responses types (e.g. auth created by plugin status for the PaymentPluginApi)? This would give us generic metrics for all plugins (both JRuby and Java) by default.

Additionally, in the base plugin, we can add helpers (e.g. `withMetrics`) for the plugin to add specific metrics relevant to that plugin (e.g. for Adyen, we would want the rate of notifications).